tcms/testplans/tests/test_edit_plan.py

```diff
@@ -0,0 +1,85 @@
+from http import HTTPStatus
+
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from tcms.tests import user_should_have_perm, BasePlanCase
+from tcms.tests.factories import TestPlanFactory
+
+
+class EditPlanViewTest(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.testplan_edit_url = reverse('plan-edit', args=[cls.plan.pk])
+
+        cls.testplan_edit_data = {
+            'author': cls.tester.pk,
+            'product': cls.plan.product.pk,
+            'product_version': cls.plan.product_version.pk,
+            'type': cls.plan.type.pk,
+            'name': cls.plan.name,
+            'text': '',
+
+            'email_settings-0-auto_to_plan_author': 'on',
+            'email_settings-0-auto_to_case_owner': 'on',
+            'email_settings-0-auto_to_case_default_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_plan_update': 'on',
+
+            'email_settings-0-plan': '1',
+            'email_settings-0-id': cls.plan.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        }
+
+        user_should_have_perm(cls.tester, perm='testplans.change_testplan')
+
+    def test_show_testplan_edit_page(self):
+        response = self.client.get(self.testplan_edit_url)
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.plan.name)
+        self.assertContains(response, _('Edit TestPlan'))
+        self._assert_notify_buttons(response)
+
+    def test_edit_testplan_text_field(self):
+        edit_data = self.testplan_edit_data.copy()
+        new_text = 'Edited: {0}'.format(self.plan.text)
+        edit_data['text'] = new_text
+
+        response = self.client.post(self.testplan_edit_url, data=edit_data, follow=True)
+
+        self.assertContains(response, new_text)
+        self.plan.refresh_from_db()
+        self.assertEqual(new_text, self.plan.text)
+
+    def test_try_editting_testplan_with_invalid_form(self):
+        testplan = TestPlanFactory(
```

Comment:
Added a invalid form testcase specific `testplan (id=2)` object, otherwise super classes `testplan (id=1)` instance objects `cleaned_data` overrides the invalid field(s) here;
https://github.com/kiwitcms/Kiwi/blob/508564d0fd90bcdab725684c2804126f5092f0b0/tcms/testplans/views.py#L82

which makes the form valid and it continues again with the valid form branch:
https://github.com/kiwitcms/Kiwi/blob/508564d0fd90bcdab725684c2804126f5092f0b0/tcms/testplans/views.py#L83